desc/objectives/_equilibrium.py

```diff
@@ -465,9 +509,29 @@ def build(self, eq, use_jit=True, verbose=1):
 
         """
         if self.grid is None:
-            self.grid = QuadratureGrid(
-                L=eq.L_grid, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP
-            )
+            if eq.node_pattern in [
+                "jacobi",
+                "cheb1",
+                "cheb2",
+                "ocs",
+                "linear",
+            ]:
+                warnings.warn(
```

Comment:
Not crazy about overriding the user input here. They should be able to use a weird grid if they want. Also, the way this is here the grid won't be set at all

Reply:
Fair point, I made it allow any grid now. and fixed the logic error